### ISSUE TITLE
Adding new filter in left sidebar

### ISSUE THREAD

**Issue by moosewebdev** [bullhorn/career-portal#208]
My client would like to add a filter to allow the user to show jobs by employmentType. Right now they have permenant and contract as the values but i am not sure how to tackle this. Selecting Permamanet would show all perm jobs and same for Contract - would show all contract jobs.

I though about just duplicating the category functions but there are so many of them plus additional variables. 

Any help would be appreciated. thansk again.

## Current behavior

-

## Expected behavior

-

## Steps to replicate behavior

1.

## Screenshots

__Current__

__Expected__

**Comment by charlesabarnes** [comment]
Hey so to add that filter option it would involve some work with modifying the search service, and sidebar controllers, and possibly some more items.  We have had a few variations of this question that you can view [here](https://github.com/bullhorn/career-portal/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed+search)for some guidance. let me know if you have any questions.

**Comment by moosewebdev** [comment]
Sorry I'm very new to Dev bit am learning fast. 

In the search service what exactly do I need to add. Can I just duplicate what was do e for category filer and change the variables to employmentType

**Comment by charlesabarnes** [comment]
Yes, it will be similar but a tad bit different as the categories relies on IDs instead of just using string values.

**Comment by moosewebdev** [comment]
I'm very confused where and what needs to be updated. I've added $employmentType to the search service.

But not sure next steps.

**Comment by moosewebdev** [comment]
How do i change this location piece for employmentType? I know the Employment types will be Permament and Contract.

location: (isSearch, fields) => {
                    if ('address(city,state)' !== fields) {
                        if (this.searchParams.location.length > 0) {
                            let delimiter = isSearch ? '"' : '\'';
                            let equals = isSearch ? ':' : '=';

                            let fragment = ' AND (';
                            let first = true;

                            for (let j = 0; j < this.searchParams.location.length; j++) {
                                if (!first) {
                                    fragment += ' OR ';
                                } else {
                                    first = false;
                                }

                                let location = this.searchParams.location[j];

                                let city = isSearch ? location.split('|')[0] : location.split('|')[0].replace(/'/g, '\'\'');
                                let state = location.split('|')[1];

                                fragment += '(address.city' + equals + delimiter + city + delimiter + ' AND address.state' + equals + delimiter + state + delimiter + ')';
                            }

                            return fragment + ')';
                        }
                    }

                    return '';
                },

**Comment by charlesabarnes** [comment]
Hey @moosewebdev So you do not need to change location to filter for employment type.  That function is used to make the search in the proper city, state format.   

As it is just a string search all you will need to do is query for the string that you desire without any formatting.  The sidebar controller, template, and additional pieces of the search service still needs to be modified based on your description.

**Comment by moosewebdev** [comment]
So id create a function below this for the employType and do my query in there. when you say query do you mean a api query or just a if employmentType == fields

or am i way off here, so im very new to angular and bithorn.

thanks

**Comment by charlesabarnes** [comment]
By query I mean the api query. 

That function is just constructing the parameter that gets added to the get request URL. 

For example for location it is returning 
`(address.city="city" AND address.state="state")`

All you need to do is `(employmentType="desiredEmploymentType")`

**Comment by moosewebdev** [comment]
what function do i put (employmentType="Permanent")

a new one similiar to location in requestParams?

**Comment by moosewebdev** [comment]
Thanks for your help. I have it working for the most part. Except when selecting the filter (either contract or permanent) i recieve an error in the GET

this is the get request https://public-rest3.bullhornstaffing.com/rest-services/ZSCEP/query/JobBoardPost?where=(isOpen=true)%20AND%20(employmentType=Contract)&fields=id,title,publishedCategory(id,name),customTextBlock5,customText4,correlatedCustomText10,costCenter,optionsPackage,address(city,state),employmentType,dateLastPublished,publicDescription,isOpen,isPublic,isDeleted&count=20&orderBy=-dateLastPublished&start=0

here is a link to the project: http://tfdl.com/careers/

**Comment by charlesabarnes** [comment]
So close! You just need quotes around the term. So employmentType='contract'

**Comment by moosewebdev** [comment]
Oh so close, ive been tinkering how do i add that to the fragment.

 fragment += 'employmentType' + equals + _this3.searchParams.employType[i];
